File: BatchTestTool/java/java automated tester/Test/to_be_tested/PersonRequest.java
package to_be_tested;

public class PersonRequest {

    public String name;

    public String sex;

    public String phoneNumber;

    public PersonRequest(String name, String sex, String phoneNumber) {
        this.name = validateName(name);
        this.sex = validateSex(sex);
        this.phoneNumber = validatePhoneNumber(phoneNumber);
    }

    public String validateName(String name) {
        if (name == null || name.isEmpty()) {
            return null;
        }
        if (name.length() > 33) {
            return null;
        }
        return name;
    }

    public String validateSex(String sex) {
        if (!sex.equals("Man") && !sex.equals("Woman") && !sex.equals("UGM")) {
            return null;
        }
        return sex;
    }

    public String validatePhoneNumber(String phoneNumber) {
        if (phoneNumber == null || phoneNumber.isEmpty()) {
            return null;
        }
        if (phoneNumber.length() != 11 || !phoneNumber.matches("\\d+")) {
            return null;
        }
        return phoneNumber;
    }
}

File: ClassEval_T/cpp/solution/PersonRequest.cpp
#include <string>

class PersonRequest {
public:
    PersonRequest(const std::string& name, const std::string& sex, const std::string& phoneNumber);

    std::string name;
    std::string sex;
    std::string phoneNumber;

private:
    std::string validate_name(const std::string& name);
    std::string validate_sex(const std::string& sex);
    std::string validate_phone_number(const std::string& phoneNumber);
    bool is_all_digits(const std::string& s);
};


PersonRequest::PersonRequest(const std::string& name, const std::string& sex, const std::string& phoneNumber)
    : name(validate_name(name)), sex(validate_sex(sex)), phoneNumber(validate_phone_number(phoneNumber)) {}


std::string PersonRequest::validate_name(const std::string& name) {
    if (name.empty() || name.length() > 33) {
        return "";
    }
    return name;
}


std::string PersonRequest::validate_sex(const std::string& sex) {
    if (sex != "Man" && sex != "Woman" && sex != "UGM") {
        return "";
    }
    return sex;
}


std::string PersonRequest::validate_phone_number(const std::string& phoneNumber) {
    if (phoneNumber.empty() || phoneNumber.length() != 11 || !is_all_digits(phoneNumber)) {
        return "";
    }
    return phoneNumber;
}


bool PersonRequest::is_all_digits(const std::string& s) {
    for (char c : s) {
        if (!isdigit(c)) {
            return false;
        }
    }
    return true;
}
